syntax = "proto3";

package freqsearch.v1;

option go_package = "github.com/saltfish/freqsearch/go-backend/pkg/pb/freqsearch/v1;freqsearchv1";

import "google/protobuf/timestamp.proto";

// Pagination request parameters
message PaginationRequest {
  int32 page = 1;      // Page number (1-indexed)
  int32 page_size = 2; // Items per page
}

// Pagination response metadata
message PaginationResponse {
  int32 total_count = 1;
  int32 page = 2;
  int32 page_size = 3;
  int32 total_pages = 4;
}

// Time range filter
message TimeRange {
  google.protobuf.Timestamp start = 1;
  google.protobuf.Timestamp end = 2;
}

// Job execution status
enum JobStatus {
  JOB_STATUS_UNSPECIFIED = 0;
  JOB_STATUS_PENDING = 1;
  JOB_STATUS_RUNNING = 2;
  JOB_STATUS_COMPLETED = 3;
  JOB_STATUS_FAILED = 4;
  JOB_STATUS_CANCELLED = 5;
}

// Strategy approval status in optimization loop
enum ApprovalStatus {
  APPROVAL_STATUS_UNSPECIFIED = 0;
  APPROVAL_STATUS_PENDING = 1;
  APPROVAL_STATUS_APPROVED = 2;
  APPROVAL_STATUS_REJECTED = 3;
  APPROVAL_STATUS_NEEDS_ITERATION = 4;
}

// Health check request
message HealthCheckRequest {}

// Health check response
message HealthCheckResponse {
  bool healthy = 1;
  string version = 2;
  map<string, bool> services = 3; // e.g., {"postgres": true, "rabbitmq": true}
}
